From 408763f32676fa60e74cb202486be4b17a43974e
From: Denys Vlasenko
Date: Tue, 03 May 2011 15:29:35 +0000
Subject: getaddrinfo.c: fix incorrect check for ERANGE from gethostbyaddr_r

Also, freeaddrinfo(NULL) is ok, no need to check parameted for NULL
before calling it.

Signed-off-by: Denys Vlasenko <dvlasenk@redhat.com>
---
diff --git a/libc/inet/getaddrinfo.c b/libc/inet/getaddrinfo.c
index e619eff..1a77c51 100644
--- a/libc/inet/getaddrinfo.c
+++ b/libc/inet/getaddrinfo.c
@@ -649,7 +649,7 @@ gaih_inet(const char *name, const struct gaih_service *service,
 						at2->family,
 						&th, tmpbuf, tmpbuflen,
 						&h, &herrno);
-				} while (rc == errno && herrno == NETDB_INTERNAL);
+				} while (rc == ERANGE && herrno == NETDB_INTERNAL);
 
 				if (rc != 0 && herrno == NETDB_INTERNAL) {
 					__set_h_errno(herrno);
@@ -855,7 +855,7 @@ getaddrinfo(const char *name, const char *service,
 					last_i = i;
 					if (hints->ai_family == AF_UNSPEC && (i & GAIH_OKIFUNSPEC))
 						continue;
-					if (p)
+					/*if (p) - freeaddrinfo works ok on NULL too */
 						freeaddrinfo(p);
 					return -(i & GAIH_EAI);
 				}
--
cgit v0.8.2.1
